src/documentmanager/plugin.ts

```diff
@@ -59,180 +47,184 @@ export
-          text: 'New Text File',
-          caption: 'Create a new text file'
-        }
-      ]);
-
-      // Add the command for a new notebook.
-      let newNotebookId = 'file-operations:new-notebook';
-
-      registry.add([
-        {
-          id: newNotebookId,
-          handler: () => {
-            browser.newUntitled('notebook').then(
-              contents => manager.open(contents)
-            );
-          }
-        }
-      ]);
-      palette.add([
-        {
-          id: newNotebookId,
-          args: void 0,
-          category: 'File Operations',
-          text: 'New Notebook',
-          caption: 'Create a new Jupyter Notebook'
-        }
-      ]);
-
-      // Add the command for saving a document.
-      let saveDocumentId = 'file-operations:save';
-
-      registry.add([
-        {
-          id: saveDocumentId,
-          handler: () => {
-            manager.save();
-            return true;
-          }
-        }
-      ]);
-      palette.add([
-        {
-          id: saveDocumentId,
-          args: void 0,
-          category: 'File Operations',
-          text: 'Save Document',
-          caption: 'Save the current document'
-        }
-      ]);
-
-      // Add the command for reverting a document.
-      let revertDocumentId = 'file-operations:revert';
-
-      registry.add([
-        {
-          id: revertDocumentId,
-          handler: () => {
-            manager.revert();
-            return true;
-          }
-        }
-      ]);
-      palette.add([
-        {
-          id: revertDocumentId,
-          args: void 0,
-          category: 'File Operations',
-          text: 'Revert Document',
-          caption: 'Revert the current document'
-        }
-      ]);
-
-      // Add the command for closing a document.
-      let closeDocumentId = 'file-operations:close';
-
-      registry.add([
-        {
-          id: closeDocumentId,
-          handler: () => {
-            manager.close();
-            return true;
-          }
-        }
-      ]);
-      palette.add([
-        {
-          id: closeDocumentId,
-          args: void 0,
-          category: 'File Operations',
-          text: 'Close Document',
-          caption: 'Close the current document'
-        }
-      ]);
-
-      // Add the command for closing all documents.
-      let closeAllId = 'file-operations:close-all';
-
-      registry.add([
-        {
-          id: closeAllId,
-          handler: () => {
-            manager.closeAll();
-            return true;
-          }
-        }
-      ]);
-      palette.add([
-        {
-          id: closeAllId,
-          args: void 0,
-          category: 'File Operations',
-          text: 'Close All',
-          caption: 'Close all open documents'
-        }
-      ]);
-
-      browser.widgetFactory = model => {
-        return manager.open(model);
+const documentManagerExtension = {
+  id: 'jupyter.extensions.documentManager',
+  requires: [DocumentManager, FileBrowserWidget],
```

Comment:
FilebrowserWidget should not be a service. We'll need to fix that going forward, fyi.